detect_secrets/core/secrets_collection.py

```diff
@@ -242,11 +257,29 @@ def scan_file(self, filename):
             return False
         try:
             with codecs.open(filename, encoding='utf-8') as f:
-                self._extract_secrets_from_file(f, filename)
+                self._extract_secrets_from_file(
+                    f,
+                    filename,
+                    suppress_unscannable_file_warnings,
+                    fail_on_file_unscannable,
+                )
 
             return True
         except IOError:
-            log.warning('Unable to open file: %s', filename)
+            if not suppress_unscannable_file_warnings:
+                log.warning(
+                    'Unable to open file: %s',
+                    filename,
+                )
+            if fail_on_file_unscannable:
+                log.error(
+                    '\nExiting since a file could not be opened.'
+                    '\nYou are seeing this message because '
+                    'the --fail-on-file-unscannable option is enabled.',
+                )
+                sys.exit(1)
+            else:
+                log.warning('Continuing scan...\n')
```

Comment:
Woops, this causes unnecessary logging to occur even when there's not an error reading the file:
```
.git/objects/76/f0b1e9e1556e8b8a3984289ea2e1c7cd42ad74 failed to load, and could not be scanned, because the file is not valid UTF-8.
If possible, convert this file to valid UTF-8 for it to be scanned.
Continuing scan...

Continuing scan...

.git/objects/7a/c8857a0a5f52f57fbec741f79f32e0022dc1d5 failed to load, and could not be scanned, because the file is not valid UTF-8.
If possible, convert this file to valid UTF-8 for it to be scanned.
Continuing scan...

Continuing scan...

.git/objects/84/d5bdbe79bfcfffe2ee219ec3ad1b6c054db860 failed to load, and could not be scanned, because the file is not valid UTF-8.
If possible, convert this file to valid UTF-8 for it to be scanned.
Continuing scan...

Continuing scan...

Continuing scan...

Continuing scan...

Continuing scan...

Continuing scan...

Continuing scan...

Continuing scan...

Continuing scan...

Continuing scan...

Continuing scan...

.git/objects/b9/fe78d49cdb9fa7b0b6f59eb72e175d8e359b86 failed to load, and could not be scanned, because the file is not valid UTF-8.
If possible, convert this file to valid UTF-8 for it to be scanned.
Continuing scan...

Continuing scan...

Continuing scan...

Continuing scan...

Continuing scan...

Continuing scan...

Continuing scan...

Continuing scan...

Continuing scan...

Continuing scan...

Continuing scan...

Continuing scan...

Continuing scan...
```

Reply:
Fixed by https://github.com/IBM/detect-secrets/pull/121/commits/b356ce95468fdfbe4db0018443ea3a038d69354c